extensions-contrib/kafka-emitter/src/main/java/io/druid/emitter/kafka/KafkaEmitter.java

```diff
@@ -0,0 +1,193 @@
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicLong;
+
+public class KafkaEmitter implements Emitter {
+  private static Logger log = new Logger(KafkaEmitter.class);
+
+  private final static String DEFAULT_KEY_SERIALIZER = "org.apache.kafka.common.serialization.StringSerializer";
+  private final static String DEFAULT_VALUE_SERIALIZER = "org.apache.kafka.common.serialization.StringSerializer";
+  private final static int DEFAULT_RETRIES = 3;
+  private long queueMemoryBound = 33554432L; // same with default value of kafka producer's buffer.memory
+  private final AtomicLong metricLost;
+  private final AtomicLong alertLost;
+  private final AtomicLong invalidLost;
+
+  private final KafkaEmitterConfig config;
+  private final Producer<String, String> producer;
+  private final Callback producerCallback;
+  private final ObjectMapper jsonMapper;
+  private final MemoryBoundLinkedBlockingQueue<String> metricQueue;
+  private final MemoryBoundLinkedBlockingQueue<String> alertQueue;
+  private final ScheduledExecutorService scheduler;
+
+  public KafkaEmitter(KafkaEmitterConfig config, ObjectMapper jsonMapper) {
+    this.config = config;
+    this.jsonMapper = jsonMapper;
+    this.producer = setKafkaProducer();
+    this.producerCallback = setProducerCallback();
+    this.metricQueue = new MemoryBoundLinkedBlockingQueue<>(queueMemoryBound);
+    this.alertQueue = new MemoryBoundLinkedBlockingQueue<>(queueMemoryBound);
+    this.scheduler = Executors.newScheduledThreadPool(3);
+    this.metricLost = new AtomicLong(0L);
+    this.alertLost = new AtomicLong(0L);
+    this.invalidLost = new AtomicLong(0L);
+  }
+
+  private Callback setProducerCallback() {
+    return new Callback() {
+      @Override
+      public void onCompletion(RecordMetadata recordMetadata, Exception e) {
+        if(e != null) {
+          log.debug("Event send failed [%s]", e.getMessage());
+          if(recordMetadata.topic().equals(config.getMetricTopic())) {
+            metricLost.incrementAndGet();
+          } else if (recordMetadata.topic().equals(config.getAlertTopic())) {
+            alertLost.incrementAndGet();
+          } else {
+            invalidLost.incrementAndGet();
+          }
+        }
+      }
+    };
+  }
+
+  private Producer<String, String> setKafkaProducer() {
+    Properties props = new Properties();
+    props.put(ProducerConfig.BOOTSTRAP_SERVERS_CONFIG, config.getBootstrapServers());
+    props.put(ProducerConfig.KEY_SERIALIZER_CLASS_CONFIG, DEFAULT_KEY_SERIALIZER);
+    props.put(ProducerConfig.VALUE_SERIALIZER_CLASS_CONFIG, DEFAULT_VALUE_SERIALIZER);
+    props.put(ProducerConfig.RETRIES_CONFIG, DEFAULT_RETRIES);
+    props.putAll(config.getKafkaProducerConfig());
+    queueMemoryBound = (props.containsKey(ProducerConfig.BUFFER_MEMORY_CONFIG) ?
+        Long.parseLong(props.getProperty(ProducerConfig.BUFFER_MEMORY_CONFIG)) : queueMemoryBound);
+
```

Comment:
The classes in jars in extensions directory are loaded by URLClassloader, but the KafkaProducer here use the appClassLoader to load classes in org.apache.kafka.common.config.ConfigDef parseType() and this lead to org.apache.kafka.common.config.ConfigException. That is the problem I got.

I found some code in KafkaIndexTask, in method newConsumer() such as:
`ClassLoader currCtxCl = Thread.currentThread().getContextClassLoader();
    try {
      Thread.currentThread().setContextClassLoader(getClass().getClassLoader());
    }finally {
      Thread.currentThread().setContextClassLoader(currCtxCl);
    }
`
This can solve the my problem. May I ask have you encounter this problem?